app/test/vesting_test.go

```diff
@@ -0,0 +1,936 @@
+	assert.NoError(s.T(), s.cctx.WaitForNextBlock())
+
+	// find and test a vesting account with endTime which is
+	// 	at least 20 seconds away time from now
+	for {
+		name := s.unusedAccount(DelayedVestingAccountType)
+		address := getAddress(name, s.cctx.Keyring).String()
+
+		resAccBytes, err := GetRawAccountInfo(s.cctx.GRPCClient, address)
+		assert.NoError(s.T(), err)
+
+		var vAcc vestingtypes.DelayedVestingAccount
+		err = vAcc.Unmarshal(resAccBytes)
+		assert.NoError(s.T(), err)
+
+		if vAcc.EndTime > tmtime.Now().Unix()+20 {
+			s.testDelegatingVestingAmount(name)
+			s.testClaimDelegationReward(name)
+			return
+		}
+	}
+}
+
+func (s *VestingModuleTestSuite) TestGenesisPeriodicVestingAccountsSpendableBalance() {
+	assert.NoError(s.T(), s.cctx.WaitForNextBlock())
+
+	initCoinsForGasFee := sdk.NewCoin(app.BondDenom, sdk.NewInt(initBalanceForGasFee))
+
+	for {
+		name := s.unusedAccount(PeriodicVestingAccountType)
+		address := getAddress(name, s.cctx.Keyring).String()
+
+		resAccBytes, err := GetRawAccountInfo(s.cctx.GRPCClient, address)
+		assert.NoError(s.T(), err)
+
+		var vAcc vestingtypes.PeriodicVestingAccount
+		err = vAcc.Unmarshal(resAccBytes)
+		assert.NoError(s.T(), err)
+
+		balances, err := GetAccountSpendableBalance(s.cctx.GRPCClient, address)
+		assert.NoError(s.T(), err)
+
+		vestedCoins := vAcc.GetVestedCoins(tmtime.Now())
+		expectedSpendableCoins := vestedCoins.Add(initCoinsForGasFee)
+		assert.EqualValues(s.T(),
+			expectedSpendableCoins.AmountOf(app.BondDenom).Int64(),
+			balances.AmountOf(app.BondDenom).Int64(),
+			"spendable balance must match")
+
+		// Stop testing once we hit an account with no spendable balance
+		if vestedCoins.IsZero() {
+			break
+		}
+		s.T().Log("waiting 3 seconds...")
+		time.Sleep(3 * time.Second)
+	}
+}
+
+func (s *VestingModuleTestSuite) TestGenesisPeriodicVestingAccountsDelegation() {
+	assert.NoError(s.T(), s.cctx.WaitForNextBlock())
+
+	// find and test a vesting account that has some vesting (locked) balance
+	for {
+		name := s.unusedAccount(PeriodicVestingAccountType)
+		address := getAddress(name, s.cctx.Keyring).String()
+
+		resAccBytes, err := GetRawAccountInfo(s.cctx.GRPCClient, address)
+		assert.NoError(s.T(), err)
+
+		var vAcc vestingtypes.PeriodicVestingAccount
+		err = vAcc.Unmarshal(resAccBytes)
+		assert.NoError(s.T(), err)
+
+		// 10 seconds is chosen to be on the safe side
+		if vAcc.StartTime > tmtime.Now().Unix()+10 {
+			s.testDelegatingVestingAmount(name)
+			return
+		}
+	}
+}
+
+// This test function tests delegation of a periodic vesting account that
+// has part of its allocation unlocked and part of it locked
+func (s *VestingModuleTestSuite) TestGenesisPeriodicVestingAccountsDelegationPartiallyVested() {
+	assert.NoError(s.T(), s.cctx.WaitForNextBlock())
+
+	// find and test a vesting account that has some vesting (locked) and
+	// some vested (unlocked) balance
+	for {
```

Comment:
Since in these tests, vesting accounts are created ahead of time, to test genesis, and they are time triggered, this is the only solution I could think of. If we can find a solution to avoid it, it will be great.

Reply:
fixed. Closing it